packages/editor/src/components/post-actions/actions.js

```diff
@@ -483,7 +476,13 @@ const viewPostAction = {
 	},
 	callback( posts, { onActionPerformed } ) {
 		const post = posts[ 0 ];
-		window.open( post.link, '_blank' );
+		let url = post.link;
+		if ( isPreviewingTheme() ) {
```

Comment:
Is this action, the reason you had to move previewing theme state to router?

Reply:
Also because the `useLink`/`Link` components have that as a dependency. If I move those to the `@wordpress/router`, it cannot depend on `@wordpress/edit-site`. See what I know about the `wp_theme_preview` parameter at https://github.com/WordPress/gutenberg/pull/63849#discussion_r1687870181

Reply:
Maybe in addition to the idea I shared here https://github.com/WordPress/gutenberg/pull/63849#discussion_r1687887936 the site editor can override this action.